android-components/components/support/ktx/src/test/java/mozilla/components/support/ktx/android/view/ActivityTest.kt

```diff
@@ -35,123 +34,111 @@ class ActivityTest {
 
     private lateinit var activity: Activity
     private lateinit var window: Window
-    private lateinit var layoutParams: WindowManager.LayoutParams
     private lateinit var decorView: View
     private lateinit var viewTreeObserver: ViewTreeObserver
-    private lateinit var windowInsetsCompat: WindowInsetsCompat
     private lateinit var windowInsets: WindowInsets
+    private lateinit var insetsController: WindowInsetsControllerCompat
+    private lateinit var layoutParams: WindowManager.LayoutParams
 
     @Before
     fun setup() {
         activity = mock()
         window = mock()
-        layoutParams = WindowManager.LayoutParams()
         decorView = mock()
         viewTreeObserver = mock()
-        windowInsetsCompat = mock()
         windowInsets = mock()
+        insetsController = mock()
+        layoutParams = WindowManager.LayoutParams()
 
         `when`(activity.window).thenReturn(window)
         `when`(window.decorView).thenReturn(decorView)
         `when`(window.decorView.viewTreeObserver).thenReturn(viewTreeObserver)
-        `when`(window.attributes).thenReturn(layoutParams)
-        `when`(windowInsetsCompat.toWindowInsets()).thenReturn(windowInsets)
         `when`(window.decorView.onApplyWindowInsets(windowInsets)).thenReturn(windowInsets)
+        `when`(window.attributes).thenReturn(layoutParams)
     }
 
     @Test
-    fun `check enterToImmersiveMode sets the correct flags`() {
-        activity.enterToImmersiveMode()
+    fun `WHEN enterImmersiveMode is called THEN insetsController hides system bars and sets bars behaviour and OnApplyWindowInsetsListener is set`() {
+        activity.enterImmersiveMode(insetsController)
 
-        // verify entering immersive mode
-        verify(decorView).systemUiVisibility = View.SYSTEM_UI_FLAG_FULLSCREEN
-        verify(decorView).systemUiVisibility = View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
-        // verify that the immersive mode restoration is set as expected
+        verify(insetsController).hide(WindowInsetsCompat.Type.systemBars())
+        verify(insetsController).systemBarsBehavior = WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
         verify(window.decorView).setOnApplyWindowInsetsListener(any())
```

Comment:
I don't think this test is required anymore with the tests for O & P but then I see that verifying the `insetController` fails.

Reply:
this test is the solo reason why we started the rewrite – we couldn't verify that calls on `insetsController`, since its instance was created by a static function inside the test :) sure, let's talk! pinged you about that on slack👍

Reply:
Sounds good 🙂 
Having the 3 tests for this is a bit of a code smell to me, which is as you say the reason for this change. FWIW my thoughts were that we could simplify this to have tests that only check the function gates e.g `is version 28` and `is version more than 28 `- having the generic 'any version' test for those specific `verify`s doesn't seem right to me